src/custom-elements/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,251 @@
+import { Component, Element, Event, EventEmitter, Prop, h } from '@stencil/core';
+
+@Component({
+  tag: 'chi-pagination',
+  styleUrl: 'pagination.scss',
+  scoped: true
+})
+
+export class Pagination {
+  @Element() el: HTMLElement;
+  /**
+   * To define number of pages to render
+   */
+  @Prop() pages: number;
+  /**
+   * To add First page / Last page icon buttons
+   */
+  @Prop() firstLast = false;
+  /**
+   * To render inverse styled pagination
+   */
+  @Prop() inverse = false;
+  /**
+   * To define current page of pagination
+   */
+  @Prop() currentPage = 1;
+  /**
+   * To show number of result items
+   */
+  @Prop() results = 0;
+  /**
+   * To allow use select number of items to show per page
+   */
+  @Prop() perPage = false;
+  /**
+   * To add jump to page text input
+   */
+  @Prop() goTo = false;
+  /**
+   * To define compact style of pagination
+   */
+  @Prop() compact = false;
+  /**
+   * To define aria label of pagination
+   */
+  @Prop() ariaLabel = 'Pagination';
+  /**
+   * To define size of pagination elements
+   */
+  @Prop() size = 'md';
+  /**
+   * Triggered when the user navigates to another page by clicking pagination item buttons or changing value of Page Jumper input
+   */
+  @Event() chiPageChange: EventEmitter<string>;
+  /**
+   * Triggered when the user changes items per page value
+   */
+  @Event() chiPageSizeChange: EventEmitter<string>;
+
+  distanceFromCurrent;
+  pagesArray = [];
+  pagesToRender = [];
+  l = null;
+
+  calculateDistanceFromCurrent() {
+    if (this.currentPage === 1) {
+      this.distanceFromCurrent = 4;
+    } else if (this.currentPage === 2) {
+      this.distanceFromCurrent = 3;
+    } else {
+      this.distanceFromCurrent = 2;
+    }
+  }
+
+  _goToPage(pageChange: HTMLElement) {
+    let pageToGo;
+
+    for (let cur = pageChange;
+         cur && !cur.classList.contains('chi-button-group');
+         cur = cur.parentNode as HTMLElement) {
+      if (cur.nodeName === 'A') {
+        pageToGo = cur.getAttribute('data-page');
+      }
+    }
+    this.chiPageChange.emit(pageToGo);
+  }
+
+  _jumpToPage(jumpTo: string) {
+    this.chiPageChange.emit(jumpTo);
+  }
+
+  _pageSizeChange(size: string) {
+    this.chiPageSizeChange.emit(size);
+  }
+
+  componentWillLoad() {
+    this.calculateDistanceFromCurrent()
+  }
+
+  componentWillUpdate() {
+    this.calculateDistanceFromCurrent();
+    this.pagesArray = [];
+    this.pagesToRender = [];
+    this.l = 0;
+  }
+
+  render() {
+    const addPage = (page, icon = null, state = '') => {
+      let pageToGo;
+
+      if (page) {
+        pageToGo = page;
+      } else if (!page && icon) {
```

Comment:
It's not necessary to check _page_ here since otherwise, it would enter the first branch
```suggestion
      } else if (icon) {
```